skimage/data/__init__.py

```diff
@@ -148,8 +205,20 @@ def _fetch(data_filename):
         return resolved_path
 
     # Case 3:
-    # Pooch needs to download the data. Let the image fetcher to search for our
-    # data. A ConnectionError is raised if no internet connection is available.
+    # Pooch not found.
+    if image_fetcher is None:
+        raise ModuleNotFoundError(
+            "The requested file is part of the scikit-image distribution, "
+            "but requries the installation of an optional dependency, pooch, "
+            "to be used. To install pooch, use your preffered python "
+            "package manager. Follow installation instruction found at "
+            "https://scikit-image.org/docs/stable/install.html"
```

Comment:
```suggestion
            "The requested file is part of the scikit-image distribution, "
            "but requires the installation of an optional dependency, pooch. "
            "To install pooch, use your preferred python package manager. "
            "For example: `conda install -c conda-forge pooch`, or "
            "`pip install -m pooch`."
```

Reply:
I prefer not pointing to our instructions since they don't have any pooch-specific documentation. I would find this confusing.

Reply:
Can we think a little longer term about this? We can work on adding instructions, lets make sure that we are doing good by future generations of developers.

Reply:
I see two kinds of people:

* Advanced users: will use shortcuts to get to google, and to search `pooch python` which will give them a package, that they will know to use conda/pip automatically. These don't need any instructions.
* Novice users, that will be tempted to use `sudo pip install pooch`. 

I am trying to help the novice users by pointing them to more fleshed out instructions than a one liner (that will likely not work and ensure their frustration).

Reply:
Reading over this, I strongly disagree with the instructions provided.

For example, I strongly discourage
```
conda install -c conda-forge pooch
```

I really think there should be a single place where we provide install instructions, and not adhoc like an Error message.

Reply:
I raised my concerns in:
https://github.com/scikit-image/scikit-image/issues/4672